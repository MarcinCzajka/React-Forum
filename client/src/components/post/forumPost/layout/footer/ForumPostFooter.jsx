import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Statistic, Icon } from "semantic-ui-react";
import moment from 'moment';

const ForumPostFooter = props => (
    <Statistic.Group size='mini' className='maxWidth roomStats noMargin'>
        <Statistic className='leftStatistic'>
            <p>Author: {props.authorNick}</p>
            <p>Created: {moment(props.creationDate).format('YYYY-MM-DD')}</p>
        </Statistic>
        <Statistic className='roomStat'>
            <Statistic.Value>
                <Link to={`/post/${props._id}`}>
                    <Icon name='comments outline'> {props.commentsCount}</Icon>
                </Link>
            </Statistic.Value>
        </Statistic>

        <Statistic className='roomStat'>
            <Statistic.Value><Icon name='eye'/>  {props.views}</Statistic.Value>
        </Statistic>
        

        <Statistic className='roomStat' style={{cursor:'pointer'}} onClick={props.updateUpvote} >
            <Statistic.Value>
                <Icon 
                    style={{color: (props.isLikedByUser ? 'green' : '')}} 
                    name='thumbs up' />  {props.upvotes}
            </Statistic.Value>
        </Statistic>
    </Statistic.Group>
)

ForumPostFooter.defaultProps = {
    comments: 0,
    views: 0,
    upvotes: 0
}

ForumPostFooter.propTypes = {
    _id: PropTypes.string.isRequired,
	authorNick: PropTypes.string.isRequired,
    creationDate: PropTypes.string.isRequired,
    commentsCount: PropTypes.number,
    views: PropTypes.number,
    upvotes: PropTypes.number,
    isLikedByUser: PropTypes.bool,
	updateUpvote: PropTypes.func.isRequired
}

export default ForumPostFooter