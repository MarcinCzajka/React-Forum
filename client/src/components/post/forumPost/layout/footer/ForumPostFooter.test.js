import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";
import ForumPostFooter from './ForumPostFooter';

const requiredProps = {
    _id: 'abcde12345',
    updateUpvote: ()=>{},
    authorNick: 'abcde asddf',
    creationDate: '2019-12-17T15:50:53.238Z'
}

describe('PostListPagination component', () => {
    test('renders without crashing', () => {
        shallow(<Router><ForumPostFooter {...requiredProps} /></Router>);
    });

    test('match snapshot /w only required props', () => {
        const component = create(<Router><ForumPostFooter {...requiredProps} /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot - liked by user', () => {
        const component = create(<Router><ForumPostFooter {...requiredProps} isLikedByUser={true} /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 1/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
            comments={3}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 2/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
            comments={12}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 3/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
            views={333}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 4/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
            views={1014}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 5/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
            upvotes={24}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 6/10', () => {
        const component = create(<Router><ForumPostFooter 
            {...requiredProps} 
            isLikedByUser={true} 
            upvotes={345}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 7/10', () => {
        requiredProps.creationDate = '2019-12-20T20:18:54.783Z';
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 8/10', () => {
        requiredProps.creationDate = '2019-12-18T22:36:40.643Z';
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 9/10', () => {
        requiredProps.authorNick = 'mArIaN';
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w different data 10/10', () => {
        requiredProps.authorNick = 'gdffdsf1321243423413';
        const component = create(<Router><ForumPostFooter 
            {...requiredProps}
        /></Router>);
        expect(component).toMatchSnapshot();
    });

    
})