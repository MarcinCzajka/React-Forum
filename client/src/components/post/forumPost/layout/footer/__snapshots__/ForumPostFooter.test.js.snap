// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListPagination component match snapshot /w different data 1/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 2/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 3/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      333
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 4/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      1014
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 5/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      24
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 6/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "green",
          }
        }
      />
        
      345
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 7/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-20
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 8/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-18
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 9/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      mArIaN
    </p>
    <p>
      Created: 
      2019-12-18
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w different data 10/10 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      gdffdsf1321243423413
    </p>
    <p>
      Created: 
      2019-12-18
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot /w only required props 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot - liked by user 1`] = `
<div
  className="ui mini statistics maxWidth roomStats noMargin"
>
  <div
    className="ui statistic leftStatistic"
  >
    <p>
      Author: 
      abcde asddf
    </p>
    <p>
      Created: 
      2019-12-17
    </p>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <a
        href="/post/abcde12345"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="comments outline icon"
          onClick={[Function]}
        >
           
        </i>
      </a>
    </div>
  </div>
  <div
    className="ui statistic roomStat"
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="eye icon"
        onClick={[Function]}
      />
        
      0
    </div>
  </div>
  <div
    className="ui statistic roomStat"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="value"
    >
      <i
        aria-hidden="true"
        className="thumbs up icon"
        onClick={[Function]}
        style={
          Object {
            "color": "green",
          }
        }
      />
        
      0
    </div>
  </div>
</div>
`;
